Add Avalonia 12 breaking changes page (#881)
AvaloniaUI/avalonia-docs: docs/avalonia12-breaking-changes.md

**Example:**

```diff
-<PackageReference Include="Avalonia" Version="11.3.12" />
+<PackageReference Include="Avalonia" Version="12.0.0" />
-<PackageReference Include="Avalonia.Themes.Fluent" Version="11.3.12" />
+<PackageReference Include="Avalonia.Themes.Fluent" Version="12.0.0" />
```

## `Avalonia.Diagnostics` package removed

The `Avalonia.Diagnostics` package has been removed.
The Dev Tools included with [Avalonia Accelerate](https://avaloniaui.net/accelerate) provide a better experience and should be preferred.

Remove the `Avalonia.Diagnostics` package from your projects and replace it with `AvaloniaUI.DiagnosticsSupport`.  
We recommend following our [Dev Tools documentation](https://docs.avaloniaui.net/accelerate/tools/dev-tools/getting-started) to install the Accelerate Dev Tools properly.

**Example:**

Project file:

```diff
-<PackageReference Include="Avalonia.Diagnostics" Version="11.3.12" />
+<PackageReference Include="AvaloniaUI.DiagnosticsSupport" Version="2.2.0" />
```

Application file:

```diff
-AttachDevTools();
+AttachDeveloperTools();
```

PR: [#20332](https://github.com/AvaloniaUI/Avalonia/pull/20332)

## Binding class hierarchy changes

> The binding class hierarchy has changed. Bindings defined in XAML files (e.g., `{Binding}`) should be unaffected. However, binding usages in the C# code need to change.

Reviewer: ```suggestion
The binding class hierarchy has changed. Bindings defined in XAML files (e.g., `{Binding}`) are unaffected. However, you must update binding usages in C# code.
```

```

Application file:

```diff
 AppBuilder.Configure<App>()
     .UseSkia()
+    .UseHarfBuzz()
```

PR: [#19852](https://github.com/AvaloniaUI/Avalonia/pull/19852)

## [Windows] Direct2D1 support removed

Avalonia no longer provides a Direct2D rendering backend. This package was not maintained, and did not have feature and performance parity with our Skia backend. The Skia backend is now the recommended rendering backend for all usages.

**Example:**

Project file:

```diff
-<PackageReference Include="Avalonia.Direct2D1" Version="11.3.12" />
+<PackageReference Include="Avalonia.Skia" Version="12.0.0" />
```

Application file:

```diff
 AppBuilder.Configure<App>()
-    .UseDirect2D1()
+    .UseSkia()
```

PR: [#20132](https://github.com/AvaloniaUI/Avalonia/pull/20132)

## [Android] App initialization changed

In Avalonia 12, the way an Android application and its activities are initialized has changed, allowing subsequent activities to properly use the `Application` class defined in your project.

1. Change your main activity to inherit from `AvaloniaMainActivity` (non-generic) instead of `AvaloniaMainActivity<TApp>` (generic).

> 2. Add a new class deriving from `AvaloniaAndroidApplication<TApp>` and apply the `[Android.App.Application]` attribute to it

Reviewer: ```suggestion
2. Add a new class deriving from `AvaloniaAndroidApplication<TApp>` and apply the `[Android.App.Application]` attribute to it.
```
